app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,113 @@
+class TasksController < ApplicationController
+  
+  
+  def index
+    @greeting = "HELLO, THERE!"
+    @tasks = Task.all
+  end
+  
+  
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: " + task_id.to_s
+      redirect_to root_path
+      return
+    end
+  end
+  
+  
+  def new
+    @task = Task.new
+    @task.completion_date = nil
```

Comment:
I have a suspicion that even without this line, your code will continue to work as expected.